torch/fx/passes/backends/nvfuser/compile.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict
+
+import torch
+from torch.fx import GraphModule
+from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner
+from torch.fx.passes.backends.nvfuser.operator_support import NvFuserOperatorSupport
+from torch._prims.executor import execute
+from torch.fx.experimental.proxy_tensor import DecompositionInterpreter
+from torch._decomp import decomposition_table
+
+import logging
+
+logging.basicConfig(level=logging.WARNING)
+logger = logging.getLogger(__name__)
+
+def aten_to_dtype(self, dtype: torch.dtype, **kwargs):
+    if len(kwargs) > 0 or not dtype:
+        raise RuntimeError("No support for other to.dtype() formats other than to.dtype(self, dtype)")
+    return torch._prims.convert_element_type(self, dtype)
+
+# decomposition_table currently contains both aten2aten and aten2prim decomposition
+# this is a hack to seperate them, as we only need aten2prim decomposition for nvfuser-supported aten graph lowering
```

Comment:
What if you need to chain an aten2aten decomp into an aten2prim decomp to get nvfuser going?

Reply:
If an op needs aten2aten->aten2prims decomps to work together, by our current design, aten2aten will happen at the dynamo+aot layer.

Reply:
Well, we have to be careful about CompositeImplicitAutograd aten2aten decompositions, and what we used to call torch._decomp aten2aten decompositions. If the original aten operator is a fused one, we also don't want to apply the aten2aten decomposition until we know that a compiler is going to be able to nail either